docs/reference/snapshot-restore/apis/repo-speed-test-api.asciidoc

```diff
@@ -0,0 +1,132 @@
+recorded in the {es} logs. You should verify yourself that this location has
+been cleaned up correctly. If there is still leftover data at the specified
+location then you should manually remove it.
+
+If the connection from your client to {es} is closed while the client is
+waiting for the result of the speed test then the test is cancelled.  Since a
+speed test takes a long time to complete, you may need to configure your client
+to wait for longer than usual for a response. On cancellation the speed test
+attempts to clean up the data it was writing, but it may not be able to remove
+it all. The path to the leftover data is recorded in the {es} logs. You should
+verify yourself that this location has been cleaned up correctly. If there is
+still leftover data at the specified location then you should manually remove
+it.
+
+NOTE: A speed test writes a substantial amount of data to your repository and
+then reads it back again. This consumes bandwidth on the network between the
+cluster and the repository, and storage space and IO bandwidth on the
+repository itself. You must ensure this load does not affect other users of
+these systems. Speed tests respect the repository settings
+`max_snapshot_bytes_per_sec` and `max_restore_bytes_per_sec` if available, and
+the cluster setting `indices.recovery.max_bytes_per_sec` which you can use to
+limit the bandwidth they consume.
+
+[[repo-speed-test-api-path-params]]
+==== {api-path-parms-title}
+
+`<repository>`::
+(Required, string)
+Name of the snapshot repository to test.
+
+[[repo-speed-test-api-query-params]]
+==== {api-query-parms-title}
+
+`blob_count`::
+(Optional, integer) The total number of blobs to write to the repository during
+the test. Defaults to `100`. For realistic experiments you should set this to
+at least `2000`.
+
+`concurrency`::
+(Optional, integer) The number of write operations to perform concurrently.
+Defaults to `10`.
+
+`seed`::
+(Optional, integer) The seed for the pseudo-random number generator used to
+generate the list of operations performed during the test. To repeat the same
+set of operations in multiple experiments, use the same seed in each
+experiment.
+
+`max_blob_size`::
+(Optional, <<size-units, size units>>) The maximum size of a blob to be written
+during the test. Defaults to `10mb`. For realistic experiments you should set
+this to at least `2gb`.
+
+`master_timeout`::
+(Optional, <<time-units, time units>>) Specifies the period of time to wait for
+a connection to the master node. If no response is received before the timeout
+expires, the request fails and returns an error. Defaults to `30s`.
+
+`timeout`::
+(Optional, <<time-units, time units>>) Specifies the period of time to wait for
+the test to complete. If no response is received before the timeout expires,
+the test is cancelled and returns an error. Defaults to `30s`.
+
+[role="child_attributes"]
+[[repo-speed-test-api-response-body]]
+==== {api-response-body-title}
+
+TODO
```

Comment:
Today we report the details of every read and write performed during the test. It'd probably be useful to add some higher-level summary statistics too, maybe only returning the low-level ones if `?detailed` is passed. Suggestions for the higher-level stats are up for discussion.

Reply:
Added simple accumulators in https://github.com/elastic/elasticsearch/pull/67247/commits/0e904d80ddcc2b7579fc7ab5f3b747947f493edd.